auth/store.go

```diff
@@ -76,10 +77,24 @@ type AuthStore interface {
 
 	// RoleGrant grants a permission to a role
 	RoleGrant(r *pb.AuthRoleGrantRequest) (*pb.AuthRoleGrantResponse, error)
+
+	// UsernameFromToken gets a username from the given Token
+	UsernameFromToken(token string) (string, bool)
+
+	// IsPutPermitted checks put permission of the user
+	IsPutPermitted(header *pb.RequestHeader, key string) bool
+
+	// IsRangePermitted checks range permission of the user
+	IsRangePermitted(header *pb.RequestHeader, key string) bool
+
+	// IsAuthEnabled checks authentication is enabled or not
+	IsAuthEnabled() bool
```

Comment:
Is it necessary to have enabled/disabled modes? It seems like it would be easy to forget to check this. Can there instead be a dummy pass-through/disabled auth implementation that gets swapped out when auth gets enabled?